core/src/repair/repair_service.rs

```diff
@@ -678,6 +684,63 @@ impl RepairService {
         }
     }
 
+    pub fn request_repair_for_shred_from_peer(
+        cluster_info: Arc<ClusterInfo>,
+        pubkey: Pubkey,
+        slot: u64,
+        shred_index: u64,
+        repair_socket: &UdpSocket,
+    ) {
+        let peer_repair_addr = cluster_info
+            .lookup_contact_info(&pubkey, |node| node.serve_repair(Protocol::UDP))
+            .unwrap()
+            .unwrap();
+        Self::request_repair_for_shred_from_address(
+            cluster_info,
+            pubkey,
+            peer_repair_addr,
+            slot,
+            shred_index,
+            repair_socket,
+        );
+    }
+
+    fn request_repair_for_shred_from_address(
+        cluster_info: Arc<ClusterInfo>,
+        pubkey: Pubkey,
+        address: SocketAddr,
+        slot: u64,
+        shred_index: u64,
+        repair_socket: &UdpSocket,
+    ) {
+        // Setup repair request
+        let nonce = thread_rng().gen_range(0..Nonce::MAX);
+        let identity_keypair = cluster_info.keypair();
+
+        // Create repair request
+        let header = RepairRequestHeader::new(cluster_info.id(), pubkey, timestamp(), nonce);
+        let request_proto = RepairProtocol::WindowIndex {
```

Comment:
When you create this request you need to add it to the `outstanding_requests` map, similar to how `serve_repair::repair_request` does it.

Without this I believe you will receive the shred but discard it in `window_service::verify_repair`

Reply:
Nice catch! I'll get this fixed and test to make sure shreds are making it all the way through